modules/stochastic_tools/include/state_sim/model/StateSimObjList.h

```diff
@@ -0,0 +1,277 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#ifndef STATESIMOBJLIST_H
+#define STATESIMOBJLIST_H
+
+#include <string>
+#include <iterator>
+
+/**
+ * Root for StateSim objects.  Unordered maps for Name or ID object lookup and tracking of deleted items.
+ * All StateSim object types should have a derived list class of their type.
+ */
+template <class T>
+class StateSimObjList
+{
+  //T _value;
+
+protected:
+  std::unordered_map<int, std::unique_ptr<T>> _item_list;
+  std::unordered_map<std::string, int> _name_to_id;
+  std::unordered_map<int, std::unique_ptr<T>> _deleted;
```

Comment:
unordered maps - Are you using the best data structure for your use case? Just posing the question.